core/time/new_spec.rb

```diff
@@ -475,4 +475,94 @@ def zone.local_to_utc(t)
       end
     end
   end
+
+  ruby_version_is "3.2" do
+    it "parses a time string" do
+      t = Time.utc(2020, 12, 24, 15, 56, 17)
+
+      Time.new("2020-12-24T15:56:17Z").should == t
+      Time.new("2020-12-25 00:56:17 +09:00").should == t
+      Time.new("2020-12-25 00:57:47 +09:01:30").should == t
+      Time.new("2020-12-25 00:56:17 +0900").should == t
+      Time.new("2020-12-25 00:57:47 +090130").should == t
+      Time.new("2020-12-25T00:56:17+09:00").should == t
+    end
+
+    it "accepts precission keyword argument" do
+      Time.new("2021-12-25 00:00:00.123456789876 +09:00", precision: 3).subsec.should == 0.123r
+      Time.new("2021-12-25 00:00:00.123456789876 +09:00", precision: nil).subsec.should == 0.123456789876r
+    end
```

Comment:
I would propose the following cases with (and without) `in` keyword argument:
- it returns Time in local timezone if timezone is not provided in the String argument
- it returns Time in timezone specified in the String argument (note: `Time#==` returns true if timezones are different, but `#to_i` equal)
- it returns Time in timezone specified in the String argument even if  the `in` keyword argument provided
- it returns Time in timezone specified with `in` keyword argument if timezone isn't provided in the String argument